src/common/tse.c

```diff
@@ -474,7 +469,7 @@ tse_task_prep_callback(tse_task_t *task)
 		d_list_del(&dtc->dtc_list);
 		/** no need to call if task was completed in one of the cbs */
 		gen = dtp_generation_get(dtp);
-		if (!dtp->dtp_completed) {
+		if (!atomic_load_relaxed(&dtp->dtp_completed)) {
```

Comment:
maybe @liw can take a look if the relaxed semantics is fine in this PR? thanks

Reply:
I'm not confident enough to say that they are fine. (For instance, I would expect that the `dtp_completed` check should happen before all the later statements. `atomic_load_relaxed` does not guarantee that, If I'm not mistaken. Hence I don't know if this is fine or not.) So personally I would stick to the default, sequentially consistent mode myself, unless I could prove that a more relaxed mode is fine and shows better performance numbers.

Reply:
fixed